### PR TITLE
Engg GUI:: customised bank name & output names fixes + other improvements

### LINKED ISSUE #15337: Engg GUI advanced calibration: output names fixes and improvements
When you use cropped calibration giving a list of spectra, you need to be able to give the "custom" bank name (from #14591). This could go in a text box that is only enabled when you select a custom range of spectra (sibling of the current text box where you can enter the spectra numbers). The text "Custom" can be given as factory default.

Possible issues. When you run cropped calibration, selecting either bank 1 or bank 2, the interface seems to be saving both .his files:
```
SaveOpenGenieAscii started
SaveOpenGenieAscii successful, Duration 0.04 seconds
Saved focused workspace as file: /home/fedemp/EnginX_Mantid/User/RB0000322/Calibration/ob+ENGINX_236516_North_bank.his
SaveOpenGenieAscii started
SaveOpenGenieAscii successful, Duration 0.04 seconds
Saved focused workspace as file: /home/fedemp/EnginX_Mantid/User/RB0000322/Calibration/ob+ENGINX_236516_South_bank.his
```
Check this and fix. Related: #14594.
I noticed this typo: "intgration" in a string::find in the presenter which might be related.

### PR DESCRIPTION
**Pull request by shahrozahmed1**
Customised bank name can now be set while carrying out cropped calibration along with SpectrumIDs, the parameter will set the workspace and .his file name according to provided bank name. Also enables GUI to only write out the appropriate _OpenGenie_ `.his` files and file names. The python script which is passed to plot the calibration output made compatible with personalised bank name. Pull Request also includes checks for the Customise Bank Name in the presenter test and documentation updates. 
## **To test:**
- _Code Check_
- Check _EnggDiffractionPresenterTest_ passes
### **Test 1**

Interface->Engineering Diffraction Analysis->Calibration
Enter RB Number: `123456`

Enable `Cropped Calibration`:

Calibration sample: `241391`
Bank Name: `Enable Spectrum-IDs`
- Check SpectrumIDs and Customise Bank Name text-field are disabled when Bank Name not `Enable Spectrum-IDs`

Spectrum IDs: `1-100`
Customise Bank Name: `Mantid`
Click **Calibrate** button below

Output:

Total of 5 output workspace  & two with the following name:
1.  `engggui_calibration_bank_Mantid`
2.  `engggui_difc_zero_peaks_Mantid`

Total of two plotted graph and one with the title name:
`Engg Gui Difc Zero Peaks Mantid`

In the following repository: `C:\EnginX_Mantid\User\1234\Calibration\`
You should be able to find single file name:
`ob+ENGINX_236516_Mantid_bank.his`

Open the `ob+ENGINX_236516_Mantid_bank.his` with notepad:
Search for _spec_no_ and check that the following text is found within the `.his` file:

```
   "spec_no"
    String
    "1-100"
```

---
### **Test 2**

Interface->Engineering Diffraction Analysis->Calibration
Enter RB Number: `123456`

Enable `Cropped Calibration`
Calibration sample: `241391`
Bank Name: `North`

Click **Calibrate** button below

_Once The Process Has Completed:_

Carry out **Test 1** again and the following error should not be displayed at any point:

```
The calibration calculations failed. Some input properties were not valid. See log messages for details. 
The cablibration did not finish correctly.
```

Fixes #15337 

[Release notes](http://www.mantidproject.org/ReleaseNotes_3_7_Diffraction#Engineering_Diffraction)

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the coding standards? Is it well structured with small focussed classes/methods/functions?
- [x] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
##### Functional Tests
- [x] Do changes function as described? Add comments below that describe the tests performed?
- [x] How do the changes handle unexpected situations, e.g. bad input?
- [x] Has the relevant documentation been added/updated?
- [x] Is user-facing documentation written in a user-friendly manner?
- [x] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by shahrozahmed1** [comment]
Sometimes `Cropped Calibration` while utilising `SpectrumIDs` failed, this was happening as the global variable `g_calibCropIdentifier` was not being reset to its initial property value: `SpectrumNumbers`.

The calibration should no longer fail and the following error message displayed should not be displayed while carrying out **Test 2** : 

```
The calibration calculations failed. Some input properties were not valid. See log messages for details. 
The cablibration did not finish correctly.
```

File: MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffractionPresenter.cpp [hunk 1]
```diff
@@ -1789,8 +1797,29 @@ void EnggDiffractionPresenter::loadVanadiumPrecalcWorkspaces(
...
+    if (specNos == "North") {
+      // when north bank is selected while cropped calib
+      saveOpenGenie(curvesWSName, "1-1200", "North", vanNo);
+    } else if (specNos == "South") {
+      // when south bank is selected while cropped calib
+      saveOpenGenie(curvesWSName, "1201-2400", "South", vanNo);
+    } else {
+
+      // when spectrumIDs are provided
+      std::string CustomisedBankName = m_view->currentCalibCustomisedBankName();
```
Comment (FedeMPouzols) [inline MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffractionPresenter.cpp]: Code changes look very good. I only spotted this very minor thing: the string in this line could be const (not worth changing).

**Comment by FedeMPouzols** [comment]
This looks good. But I think that  the current behavior the name of the "customised bank" can be counterintuitive to users. If the user doesn't give any value the default `cropped` is used, but this is not shown (the lineEdit `lineEdit_cropped_customise_bank_name` is not set to any default). This could be done on start-up, giving a factory default value to `user-params-calib-cropped-customise-nam` in `readSettings()`.

I also noticed that the interface is not outputting any special log when saving the `.his` files? If so, it would be good to add a small log message (notice level maybe) saying that the file has been saved and where (as when the calibration produces the `.prm` files and says "Calibration file written as C:/...".

File: MantidQt/CustomInterfaces/inc/MantidQtCustomInterfaces/EnggDiffraction/EnggDiffractionQtTabCalib.ui [hunk 2]
```diff
@@ -220,8 +220,71 @@
...
+        <item>
+         <widget class="QLineEdit" name="lineEdit_cropped_customise_bank_name">
+          <property name="sizePolicy">
+           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
+            <horstretch>1</horstretch>
+            <verstretch>0</verstretch>
+           </sizepolicy>
+          </property>
+          <property name="text">
+           <string>cropped</string>
```
Comment (shahrozahmed1) [inline MantidQt/CustomInterfaces/inc/MantidQtCustomInterfaces/EnggDiffraction/EnggDiffractionQtTabCalib.ui]: @FedeMPouzols default value for `Customise Bank Name` as `cropped` should be there for first time use, but I have enabled it to remember the previous `Customise Bank Name` which is set by user. However I can change this and get it to return `cropped` as default value every time the GUI is opened?

File: MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffractionPresenter.cpp [hunk 3]
```diff
@@ -2282,8 +2321,8 @@ void EnggDiffractionPresenter::saveOpenGenie(const std::string inputWorkspace,
             " Please check also the log messages for details.";
     throw;
   }
-  g_log.notice() << "Saved focused workspace as file: " << saveDir.toString()
-                 << std::endl;
```
Comment (shahrozahmed1) [inline MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffractionPresenter.cpp]: @FedeMPouzols It seems that log message was always being outputted when file was saved but not the best log message hence the confusion was caused probably.

**Comment by FedeMPouzols** [comment]
Works well. I tested it following the point-by-point testing instructions and also additional conditions.
The files are generated correctly and with the proper names. The plots also look good.
Code changes are good, the tests have been updated and pass.

Also in a couple of places, "spectrum IDs" has been replaced with "Spectrum numbers" which is the proper term for it. Only last second I realized that in the "cropped calibration" box we still have "spectrum IDs" in the combobox "Enable Spectrum-IDs". I'd call it "Use spectrum numbers" or similar.
@shahrozahmed1 : can you do that small change, so the interface doesn't use the evil and confusing wording "spectrum ID"?

**Comment by FedeMPouzols** [comment]
The release note for this PR had been written in the old wiki system and has already migrated to the new RST release note pages.

**Comment by FedeMPouzols** [comment]
This looks good. Summary for the gatekeeper:
-  I tested it following the point-by-point testing instructions and also additional combinations of focus and calibrate actions.
- The files are generated correctly and with the proper names. The plots also look good.
- Code changes are good, the tests have been updated and pass.
- The evil "spectrum ID" wording has been corrected in the GUI. Another issue follows to correct it in all the source files (variable names, widget names, etc.)
- The release note for this PR had been written in the old wiki system and was already migrated to the new RST release note pages.

:shipit:

**Pull request merged**